_posts/2023-07-17-13.0.0-release.md

```diff
@@ -0,0 +1,215 @@
+
+A long-standing issue where S3 support could crash at shutdown because of resources
+still being alive after S3 finalization has been fixed (GH-36346). Now, attempts
+to use S3 resources (such as making filesystem calls) after S3 finalization should
+result in a clean error.
+
+The GCS filesystem accepts a new option to set the project id (GH-36227).
+
+### IPC
+
+Nullability and metadata information for sub-fields of map types is now preserved
+when deserializing Arrow IPC (GH-35297).
+
+### Orc
+
+The Orc adapter now maps Arrow field metadata to Orc type attributes when writing,
+and vice-versa when reading (GH-35304).
+
+### Parquet
+
+It is now possible to write additional metadata while a `ParquetFileWriter` is
+open (GH-34888).
+
+Writing a page index can be enabled selectively per-column (GH-34949).
+In addition, page header statistics are not written anymore if the page
+index is enabled for the given column (GH-34375), as the information would
+be redundant and less efficiently accessed.
+
+Parquet writer properties allow specifying the sorting columns (GH-35331).
+The user is responsible for ensuring that the data written to the file
+actually complies with the given sorting.
+
+CRC computation has been implemented for v2 data pages (GH-35171).
+It was already implemented for v1 data pages.
+
+Writing compliant nested types is now enabled by default (GH-29781). This
+should not have any negative implication.
+
+Attempting to load a subset of an Arrow extension type is now forbidden
+(GH-20385). Previously, if an extension type's storage is nested (for example
+a "Point" extension type backed by a `struct<x: float64, y: float64>`),
+it was possible to load selectively some of the columns of the storage type.
+
+### Substrait
+
+Support for various functions has been added: "stddev", "variance", "first",
+"last" (GH-35247, GH-35506).
+
+Deserializing sorts is now supported (GH-32763). However, some features,
+such as clustered sort direction or custom sort functions, are not
+implemented.
+
+### Miscellaneous
+
+`FieldRef` sports additional methods to get a flattened version of nested
+fields (GH-14946). Compared to their non-flattened counterparts,
+the methods `GetFlattened`, `GetAllFlattened`, `GetOneFlattened` and
+`GetOneOrNoneFlattened` combine a child's null bitmap with its ancestors'
+null bitmaps such as to compute the field's overall logical validity bitmap.
+
+In other words, given the struct array `[null, {'x': null}, {'x': 5}]`,
+`FieldRef("x")::Get` might return `[0, null, 5]`
+while `FieldRef("y")::GetFlattened` will *always* return `[null, null, 5]`.
+
+`Scalar::hash()` has been fixed for sliced nested arrays (GH-35360).
+
+A new floating-point to decimal conversion algorithm exhibits much better
+precision (GH-35576).
+
+It is now possible to cast between scalars of different list-like types
+(GH-36309).
+
+## C# notes
+
+## Go notes
+
+## Java notes
+
+## JavaScript notes
+
+## Python notes
```

Comment:
```suggestion
## Python notes

Compatibility notes:

* The default format version for Parquet has been bumped from 2.4 to 2.6 [GH-35746](https://github.com/apache/arrow/issues/35746)
* Support for Python 3.7 is dropped [GH-34788](https://github.com/apache/arrow/issues/34788)

New features:

* Conversion to non-nano datetime64 for pandas >= 2.0 is now supported [GH-33321](https://github.com/apache/arrow/issues/33321)
* Write page index is now supported [GH-36284](https://github.com/apache/arrow/issues/36284)
* Bindings for reading JSON format in Dataset are added [GH-34216](https://github.com/apache/arrow/issues/34216)
* `keys_sorted` property of MapType is now exposed [GH-35112](https://github.com/apache/arrow/issues/35112)

Other improvements:

* Common python functionality between `Table` and `RecordBatch` classes has been consolidated ( [GH-36129](https://github.com/apache/arrow/issues/36129), [GH-35415](https://github.com/apache/arrow/issues/35415), [GH-35390](https://github.com/apache/arrow/issues/35390), [GH-34979](https://github.com/apache/arrow/issues/34979), [GH-34868](https://github.com/apache/arrow/issues/34868), [GH-31868](https://github.com/apache/arrow/issues/31868))
* Some functionality for `FixedShapeTensorType` has been improved (`__reduce__` [GH-36038](https://github.com/apache/arrow/issues/36038), picklability [GH-35599](https://github.com/apache/arrow/issues/35599))
* Pyarrow scalars can now be accepted in the `array` constructor [GH-21761](https://github.com/apache/arrow/issues/21761)
* DataFrame Interchange Protocol implementation and usage is now documented [GH-33980](https://github.com/apache/arrow/issues/33980)
* Conversion between Arrow and Pandas for map/pydict now has enhanced support [GH-34729](https://github.com/apache/arrow/issues/34729)
* Usability of `pc.map_lookup` / `MapLookupOptions` is improved [GH-36045](https://github.com/apache/arrow/issues/36045)
* `zero_copy_only` keyword can now also be accepted in `ChunkedArray.to_numpy()` [GH-34787](https://github.com/apache/arrow/issues/34787)
* Python C++ codebase now has linter support in Archery and the CI [GH-35485](https://github.com/apache/arrow/issues/35485)

Relevant bug fixes:

* `__array__` numpy conversion for Table and RecordBatch is now corrected so that `np.asarray(pa.Table)` doesn’t return a transposed result [GH-34886](https://github.com/apache/arrow/issues/34886)
* `parquet.write_to_dataset` doesn't create empty files for non-observed dictionary (category) values anymore [GH-23870](https://github.com/apache/arrow/issues/23870)
* Dataset writer now also correctly follows default Parquet version of 2.6 [GH-36537](https://github.com/apache/arrow/issues/36537)
* Comparing `pyarrow.dataset.Partitioning` with other type is now correctly handled [GH-36659](https://github.com/apache/arrow/issues/36659)
* Pickling of pyarrow.dataset PartitioningFactory objects is now supported [GH-34884](https://github.com/apache/arrow/issues/34884)
* None schema is now disallowed in parquet writer [GH-35858](https://github.com/apache/arrow/issues/35858)
* `pa.FixedShapeTensorArray.to_numpy_ndarray` is not failing on sliced arrays [GH-35573](https://github.com/apache/arrow/issues/35573)
* Halffloat type is now supported in the conversion from Arrow list to pandas [GH-36168](https://github.com/apache/arrow/issues/36168)
* `__from_arrow__` is now also implemented for `Array.to_pandas` for pandas extension data types [GH-36096](https://github.com/apache/arrow/issues/36096)
```